### ISSUE TITLE
[feature] conan config install a single file

### ISSUE THREAD

**Issue by kkaja123** [conan-io/conan#6207]
When using `conan config install`, allow a single configuration file to be installed. 

Normally, the command requires a git repository, a local folder, or a zip file that contains the file(s) to be installed. If I have a single file I want installed, I would have to put it into a folder or zip it into a file.

The current result:
```
conan config install my_settings\settings.yml
Traceback (most recent call last):
  File "C:\Users\Kyle.Kaja\AppData\Roaming\Python\Python37\site-packages\conans\client\command.py", line 1859, in run
    method(args[0][1:])
  File "C:\Users\Kyle.Kaja\AppData\Roaming\Python\Python37\site-packages\conans\client\command.py", line 554, in config
    target_folder=args.target_folder)
  File "C:\Users\Kyle.Kaja\AppData\Roaming\Python\Python37\site-packages\conans\client\conan_api.py", line 78, in wrapper
    return f(*args, **kwargs)
  File "C:\Users\Kyle.Kaja\AppData\Roaming\Python\Python37\site-packages\conans\client\conan_api.py", line 611, in config_install
    source_folder=source_folder, target_folder=target_folder)
  File "C:\Users\Kyle.Kaja\AppData\Roaming\Python\Python37\site-packages\conans\client\conf\config_installer.py", line 235, in configuration_install
    _process_config(config, cache, output, requester)
  File "C:\Users\Kyle.Kaja\AppData\Roaming\Python\Python37\site-packages\conans\client\conf\config_installer.py", line 190, in _process_config
    _process_zip_file(config, config.uri, cache, output, tmp_folder)
  File "C:\Users\Kyle.Kaja\AppData\Roaming\Python\Python37\site-packages\conans\client\conf\config_installer.py", line 63, in _process_zip_file
    unzip(zippath, tmp_folder, output=output)
  File "C:\Users\Kyle.Kaja\AppData\Roaming\Python\Python37\site-packages\conans\client\tools\files.py", line 104, in unzip
    with zipfile.ZipFile(filename, "r") as z:
  File "C:\Program Files\Python37\lib\zipfile.py", line 1225, in __init__
    self._RealGetContents()
  File "C:\Program Files\Python37\lib\zipfile.py", line 1292, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file

ERROR: File is not a zip file
```

This feature is useful in the case where you only want to install specific files from a single folder.
For example,
```
my_settings\
  settings.yml
  remotes.txt

conan config install my_settings  (Installs settings.yml and remotes.txt)
conan config install my_settings\settings.yml  (Installs settings.yml and ignores remotes.txt)
```

It is also useful when the current folder structure does not facilitate creating a folder specifically for configuration files (e.g. a git repository).
```
my_git_repo\
  settings.yml
  unrelated.txt
  ...

conan config install .\
Installing settings.yml
Copying file unrelated.txt to C:\Users\Kyle.Kaja\.conan\.  (I don't want this installed here!!)
...

OR

mkdir temp_settings
copy settings.yml temp_settings
conan config install temp_settings
Installing settings.yml
rmdir temp_settings /s  (Remove the directory so Git doesn't try to track that new file.)
(So much to do when all I want to do is install a single configuration file.)

PREFERRED

conan config install settings.yml  (Ahh! Simple, short, and intuitive!)
Installing settings.yml
```

**Comment by memsharded** [comment]
Hi @kkaja123 

I think this can make sense and shouldn't be very difficult to implement. The only thing is that our backlog is already totally full, so will label it accordingly and lets see if this feature gets contributed by the community. Thanks for the suggestion.

**Comment by sagarafr** [comment]
Hi,
I would like to contribute to this feature.

**Comment by memsharded** [comment]
Hi @sagarafr 

Great! Assigned to you. Don't hesitate to ask if you have any question, good luck.

**Comment by memsharded** [comment]
Implemented in https://github.com/conan-io/conan/pull/7840, will be available in Conan 1.31
